crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -196,6 +196,24 @@ impl EventCache {
         Ok(())
     }
 
+    /// Try to find an event by id in all the rooms.
+    ///
+    /// Note: this does a linear scan, so it could be slow. If performance
+    /// requires it, using a direct mapping of event id -> event might be
```

Comment:
Does this tell the user that they should create such a mapping or that we, the developers, will provide it if this method turns out to be slow?

Reply:
You're right, it's unclear. We should do that, if needs be; i'll convert this into a code non-doc comment.